### ISSUE TITLE
%pastebin returns an HTTP link when the underlying service supports HTTPS

### ISSUE THREAD

**Issue by illright** [ipython/ipython#12968]
## Steps to reproduce

* Execute `%pastebin 1`

## Expected result
An HTTPS link to the paste result is returned

## Actual result
An HTTP link to the paste result is returned

## System info
Manjaro Linux, Python 3.9.4, IPython 7.22.0 (reproduces on IPython 8.0.0.dev)

**Comment by Lightyagami1** [comment]
Hi I would like to take a stab at this.
looking at the code 
```
    @line_magic
    def pastebin(self, parameter_s=''):
        """Upload code to dpaste.com, returning the URL.

        Usage:\\
          %pastebin [-d "Custom description"] 1-7

        The argument can be an input history range, a filename, or the name of a
        string or macro.

        Options:

          -d: Pass a custom description. The default will say
              "Pasted from IPython".
        """
        opts, args = self.parse_options(parameter_s, 'd:')

        try:
            code = self.shell.find_user_code(args)
        except (ValueError, TypeError) as e:
            print(e.args[0])
            return

        post_data = urlencode(
            {
                "title": opts.get("d", "Pasted from IPython"),
                "syntax": "python",
                "content": code,
            }
        ).encode("utf-8")

        request = Request(
            "http://dpaste.com/api/v2/",
            headers={"User-Agent": "IPython v{}".format(version)},
        )
        response = urlopen(request, post_data)
        return response.headers.get('Location')
```

the http is hard coded into  endpoint of `dpaste.com/api/v2/` in the code.
just changing `http` to `https` worked fine in my quick testing.

just wanted to ask, should opting for http or https be based on some parameter instead rather hardcoding the url.

**Comment by MrMino** [comment]
@Lightyagami1 hardcoding the proto into the url is ok. There's no other option for it in there.

In truth, I haven't done it myself only so that someone else can come in, add this simple change, and get their first PR in :).

**Comment by MrMino** [comment]
Fixed in #13056
